packages/financial-templates-lib/test/clients/ExpiringMultiPartyClient.js

```diff
@@ -1,621 +1,943 @@
+    throw new Error(
+      `Contract versioned specified ${supportedVersions} is not part of the supported contracts for this test suit`
+    );
+  }
+  // Return if the `currentTestIterationVersion` is part of the supported versions includes any. Returning true
+  // means that test will be run. Else, if returned false, the test will be skipped.
+  return supportedVersions.includes(currentTestIterationVersion) || supportedVersions.includes("any");
+};
+
+const _createConstructorParamsForContractVersion = async function(contractVersion, contractType) {
+  let constructorParams = {
+    expirationTimestamp: (await timer.getCurrentTime()).toNumber() + 100000,
+    withdrawalLiveness: "1000",
+    collateralAddress: collateralToken.address,
+    tokenAddress: syntheticToken.address,
+    finderAddress: finder.address,
+    priceFeedIdentifier: padRight(utf8ToHex(identifier), 64),
+    liquidationLiveness: "1000",
+    collateralRequirement: { rawValue: toWei("1.5") },
+    disputeBondPercentage: { rawValue: toWei("0.1") },
+    sponsorDisputeRewardPercentage: { rawValue: toWei("0.1") },
+    disputerDisputeRewardPercentage: { rawValue: toWei("0.1") },
+    minSponsorTokens: { rawValue: convertSynthetic("5") },
+    timerAddress: timer.address,
+    excessTokenBeneficiary: store.address,
+    financialProductLibraryAddress: ZERO_ADDRESS
+  };
+
+  if (contractVersion == "1.2.2") {
+    constructorParams.disputerDisputeRewardPct = constructorParams.disputerDisputeRewardPercentage;
+    constructorParams.sponsorDisputeRewardPct = constructorParams.sponsorDisputeRewardPercentage;
+    constructorParams.disputeBondPct = constructorParams.disputeBondPercentage;
+  }
 
-        emp = await ExpiringMultiParty.new(constructorParams);
-        await syntheticToken.addMinter(emp.address);
-        await syntheticToken.addBurner(emp.address);
-
-        client = new ExpiringMultiPartyClient(
-          dummyLogger,
-          ExpiringMultiParty.abi,
-          web3,
-          emp.address,
-          testConfig.collateralDecimals,
-          testConfig.syntheticDecimals,
-          testConfig.priceFeedDecimals
-        );
-        await collateralToken.approve(emp.address, convertCollateral("1000000"), { from: sponsor1 });
-        await collateralToken.approve(emp.address, convertCollateral("1000000"), { from: sponsor2 });
+  if (contractType == "Perpetual") {
+    configStore = await getTruffleContract("ConfigStore", web3, contractVersion).new(
+      {
+        timelockLiveness: 86400, // 1 day
+        rewardRatePerSecond: { rawValue: "0" },
+        proposerBondPercentage: { rawValue: "0" },
+        maxFundingRate: { rawValue: toWei("0.00001") },
+        minFundingRate: { rawValue: toWei("-0.00001") },
+        proposalTimePastLimit: 0
+      },
+      timer.address
+    );
+
+    fundingRateIdentifier = web3.utils.utf8ToHex("TEST_FUNDiNG_IDENTIFIER");
```

Comment:
```suggestion
    fundingRateIdentifier = web3.utils.utf8ToHex("TEST_FUNDING_IDENTIFIER");
```

Reply:
good catch. have fixed this in other places as well.